File: Habitica/src/main/java/com/habitrpg/android/habitica/data/TutorialRepository.java
package com.habitrpg.android.habitica.data;

import com.habitrpg.android.habitica.models.TutorialStep;

import java.util.List;

import io.realm.RealmResults;
import rx.Observable;

public interface TutorialRepository extends BaseRepository {

    Observable<TutorialStep> getTutorialStep(String key);
    Observable<RealmResults<TutorialStep>> getTutorialSteps(List<String> keys);

}

File: Habitica/src/main/java/com/habitrpg/android/habitica/data/TutorialRepository.kt
package com.habitrpg.android.habitica.data

import com.habitrpg.android.habitica.models.TutorialStep
import kotlinx.coroutines.flow.Flow

interface TutorialRepository : BaseRepository {

    fun getTutorialStep(key: String): Flow<TutorialStep>
    fun getTutorialSteps(keys: List<String>): Flow<out List<TutorialStep>>
}
